server/controllers/userController.js

```diff
@@ -112,25 +112,65 @@ class UserControllers {
    * @memberof UserControllers
    */
   static resetPassword(req, res) {
-    const { email } = req.body;
-    pool.connect((err, client, done) => {
-      if (err) throw err;
-      const query = 'SELECT * FROM users WHERE email=$1';
-      const value = [email];
-      client.query(query, value, (error, result) => {
-        done();
-        if (error || result.rowCount === 0) {
-          return res.status(404).json({ status: 404, error: 'There is no account associated with this email' });
-        }
-        res.status(200).json({
-          status: 200,
-          data: [{
-            message: 'Check your email for password reset link',
-            email: result.rows[0].email
-          }]
+    try {
+      const { email } = req.body;
+      pool.connect((err, client, done) => {
+        if (err) throw err;
+        const query = 'SELECT * FROM users WHERE email=$1';
+        const value = [email];
+        client.query(query, value, (error, result) => {
+          done();
+          if (error || result.rowCount === 0) {
+            return res.status(404).json({ status: 404, error: 'There is no account associated with this email' });
+          }
+          res.status(200).json({
+            status: 200,
+            data: [{
+              message: 'Check your email for password reset link',
+              email: result.rows[0].email
+            }]
+          });
+        });
+      });
+    } catch (error) {
+      return res.status(500).json({ status: 500, error: 'Server error' });
+    }
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns
+   * @memberof UserControllers
+   */
+  static contestInElection(req, res) {
+    try {
+      const id = Number(req.params.user_id);
```

Comment:
I need you to validate that it is only admin that make this API call

Reply:
Yes, I was thinking of creating all the endpoints but securing routes.
Is it actually okay with you?

Reply:
What after creating all the endpoint and no time to secure it. While finishing all the endpoint is important to meet expectation, I recommend you secure it because it is better you did not finish all route and the ones you have is secured and validated than having all the route without a proper logic